clang/test/AST/Interp/records.c

```diff
@@ -0,0 +1,114 @@
+// UNSUPPORTED: asserts
+// REQUIRES: asserts
+// ^ this attempts to say "don't actually run this test", because it's broken
```

Comment:
It's broken even with the changes in this PR?

Reply:
Yeah, the test was meant to validate the "iterating the init expressions in the init list in-order will work for C records", but it doesn't get that far: in trying to do so, I uncovered a different bug.

What do you usually do, when that happens? Stop and dig out the bug? Or do you have some way you're deferring/keeping track of those kinds of things?

Reply:
In this case, I would fix the other bug first.

Reply:
Sounds good, I'll take a look at it when I get a chance.